server/subscribe.go

```diff
@@ -265,6 +267,86 @@ func (p *Plugin) editChannelSubscription(modifiedSubscription *ChannelSubscripti
 	})
 }
 
+func inAllowedGroup(inGroups []*jira.UserGroup, allowedGroups []string) bool {
+	for _, inGroup := range inGroups {
+		for _, allowedGroup := range allowedGroups {
+			if strings.TrimSpace(inGroup.Name) == strings.TrimSpace(allowedGroup) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// hasPermissionToManageSubscription checks if MM user has permission to manage subscriptions in given channel.
+// returns nil if the user has permission and a discriptive error otherwise.
+func (p *Plugin) hasPermissionToManageSubscription(userId, channelId string) error {
+	cfg := p.getConfig()
+
```

Comment:
If the user has higher admin permissions (aka sysadmin) should it be allowed to proceed?

Reply:
IMO, Yes.  If sysadmin AND a proper JIRA user, then they can set it.

Reply:
Yes. The permissions checking functions take care of that.